### ISSUE TITLE
Сan't find referenced class ClassName$DefaultImpls

### ISSUE THREAD

**Issue by tox1cozZ** [Guardsquare/proguard#290]
**Proguard 7.3.0-beta2**
**Kotlin 1.7.20**

I don't understand why he is looking for the **DefaultImpls** class for the annotation?
And in general, I have Java 8 and there is a parameter _-Xjvm-default=all_ in the compiler arguments

![Screenshot_6](https://user-images.githubusercontent.com/7694196/199303463-68bff5bb-bf70-418a-b668-aad5ba3f960f.jpg)
![Screenshot_7](https://user-images.githubusercontent.com/7694196/199303484-f7720c6e-3dca-4dda-b8bc-93cc5c4f5a12.jpg)

**Comment by oliver-charlesworth** [comment]
I can reproduce this symptom when I use `keepkotlinmetadata`.

**Repro**

*src/main/kotlin/wat/Foo.kt*

```
package wat

interface Foo
```

*build.gradle.kts*

```
import proguard.gradle.ProGuardTask

buildscript {
    dependencies {
        classpath("com.guardsquare:proguard-gradle:7.3.0-beta2")
    }
}

plugins {
    kotlin("jvm") version "1.7.21"
}

repositories {
    mavenCentral()
}

val proguard by tasks.registering(ProGuardTask::class) {
    injars(tasks.jar)
    outjars(buildDir.resolve("obfuscated/obfuscated.jar"))

    libraryjars(configurations.compileClasspath)

    val javaHome = System.getProperty("java.home")
    libraryjars(
        mapOf(
            "jarfilter" to "!**.jar",
            "filter"    to "!module-info.class",
        ),
        "$javaHome/jmods/java.base.jmod"
    )

    keepkotlinmetadata()
    dontshrink()
    dontoptimize()
    printmapping()
}
```

*Error*

```
ProGuard, version 7.3.0-beta2
Warning: wat.Foo: can't find referenced class wat.Foo$DefaultImpls
```

If I remove the `keepkotlinmetadata()`, then this succeeds.

**Comment by mrjameshamilton** [comment]
Thanks for the sample @oliver-charlesworth ! We tracked down the cause solved the underlying issue in [ProGuardCORE](https://github.com/Guardsquare/proguard-core/commit/0a000b9000a5c4586915f9a2cca1c3696de9c1dc). This should make it into the next patch release of ProGuard.

For now, you should be able to add `-dontwarn` for those classes (e.g. `dontwarn("**DefaultImpls")`), or use `-ignorewarnings`.

**Comment by mrjameshamilton** [comment]
Hi @tox1cozZ , @oliver-charlesworth ! The fix is now in ProGuard 7.3.1.
